### ISSUE TITLE
Help on error

### ISSUE THREAD

**Issue by shersoni610** [robgon-art/open-clip#1]
hello,

I am trying to run the examples on Apple M1 (cpu). I get the following error:

ValueError: Batch dimension of `input_ids` should be 18, but is 6.

Any help?

Thanks

**Comment by robgon-art** [comment]
Hello, thank you for your interest in this project. Yes, I can help track down the source of the error.

Please send some more info about the problem.

Which Colab(s) show the error?

1. [Test CLIP with Library of Congress Images](https://colab.research.google.com/github/robgon-art/open-clip/blob/main/Test_LOC_Photos_with_CLIP.ipynb)
2. [Test OpenCLIP with Library of Congress Images](https://colab.research.google.com/github/robgon-art/open-clip/blob/main/Test_LOC_Photos_with_OpenCLIP.ipynb)
3. [Index Library of Congress Photos with OpenCLIP](https://colab.research.google.com/github/robgon-art/open-clip/blob/main/Index_LOC_Photos_with_OpenCLIP.ipynb)
4. [Search for Library of Congress Photos with OpenCLIP](https://colab.research.google.com/github/robgon-art/open-clip/blob/main/Search_for_LOC_Photos_with_OpenCLIP.ipynb)
5. [Create Captions with CoCa and OpenCLIP](https://colab.research.google.com/github/robgon-art/open-clip/blob/main/Create_Captions_with_CoCa_and_OpenCLIP.ipynb)

Which code block(s) throws the error? Please send the first line in the code block where it fails to help me track it down. For example:
- Test_LOC_Photos_with_CLIP.ipynb: image_features /= image_features.norm(dim=-1, keepdim=True)

And finally, please copy and paste in all of the output from the failing code block(s).

Cheers,
-Rob

**Comment by robgon-art** [comment]
Hi, I ran the code on Colab, and answered my own questions:

`- Create_Captions_with_CoCa_and_OpenCLIP.ipynb: new_captions = []`

```ValueError                                Traceback (most recent call last)
[<ipython-input-6-70b6b4fdc2ac>](https://localhost:8080/#) in <cell line: 2>()
      4   im = transform(im).unsqueeze(0).cuda()
      5   with torch.no_grad(), torch.cuda.amp.autocast():
----> 6     generated = model.generate(im)
      7   caption = open_clip.decode(generated[0]).split("<end_of_text>")[0].replace("<start_of_text>", "")[:-2]
      8   new_captions.append(caption)

1 frames
[/usr/local/lib/python3.10/dist-packages/open_clip/coca_model.py](https://localhost:8080/#) in generate(self, image, text, seq_len, max_seq_len, temperature, generation_type, top_p, top_k, pad_token_id, eos_token_id, sot_token_id, num_beams, num_beam_groups, min_seq_len, stopping_criteria, repetition_penalty, fixed_output_length)
    211 
    212             if generation_type == "beam_search":
--> 213                 output = self._generate_beamsearch(
    214                     image_inputs = image,
    215                     pad_token_id=pad_token_id,

[/usr/local/lib/python3.10/dist-packages/open_clip/coca_model.py](https://localhost:8080/#) in _generate_beamsearch(self, image_inputs, pad_token_id, eos_token_id, sot_token_id, num_beams, num_beam_groups, min_seq_len, stopping_criteria, logit_processor, logit_warper)
    329 
    330         if num_beams * batch_size != batch_beam_size:
--> 331             raise ValueError(
    332                 f"Batch dimension of `input_ids` should be {num_beams * batch_size}, but is {batch_beam_size}."
    333             )

ValueError: Batch dimension of `input_ids` should be 18, but is 6.
```

I'll see if I can fix it.
Cheers

**Comment by robgon-art** [comment]
OK, this seems to be a known problem with a recent update to transformers.

There is a discussion here: https://github.com/mlfoundations/open_clip/issues/575

The suggested fix is going back to transformers==4.30.2, so I did that in my colab, and it seems to work fine now.

The fix is at the top:
`!pip install open_clip_torch transformers==4.30.2`

Please try this and let me know if it works.

**Comment by robgon-art** [comment]
Closing this for now. But if it’s still a problem, feel free to reopen it.
